contracts/SmartOrderStrategy.sol

```diff
@@ -79,15 +84,20 @@ contract SmartOrderStrategy is ISmartOrderStrategy, AdminManagement {
             }
 
             // calculate input amount if ratio should be applied
-            if (_inputRatio != Constant.BPS_MAX) {
-                inputTokenBalance = (inputTokenBalance * _inputRatio) / Constant.BPS_MAX;
+            // we provide a _numerator and a _denominator to decide a ratio
+            if (_numerator != _denominator) {
+                inputTokenBalance = (inputTokenBalance * _numerator) / _denominator;
```

Comment:
Add a test for `_denominator` being `0` while `_numerator > 0`

Reply:
ACK